app/src/main/java/com/nickytoolchick/agraph/MainActivity.kt

```diff
@@ -16,22 +16,33 @@ import kotlinx.serialization.json.Json
 class MainActivity : AppCompatActivity() {
 
     private var chartOptions = ChartOptions()
+    private var datasetOptions = DatasetOptions()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
         binding.configureChartButton.setOnClickListener {
-            val configIntent = Intent(this, ChartOptionsActivity::class.java)
-            configIntent.putExtra("stableChartOptions", Json.encodeToString(chartOptions))
-            resultLauncher.launch(configIntent)
+            val charOptionsIntent = Intent(this, ChartOptionsActivity::class.java)
+            charOptionsIntent.putExtra("stableChartOptions", Json.encodeToString(chartOptions))
```

Comment:
Do not use hardcoded identifiers.

Transferring an extra data identifier to a separate Enum class or object can be a good practice. This can ensure the correctness of the identifier and help avoid typos or errors in the use of identifiers for the correct operation of the program

* Example:
```kotlin
object Exstras{
val STABLE_DATABASE_OPTIONS = "stableDatasetOptions"
}
```